apollo-router/src/warp_http_server_factory.rs

```diff
@@ -229,6 +231,19 @@ fn redirect_to_studio(host: Option<Authority>) -> Box<dyn Reply> {
     }
 }
 
+fn get_health_request() -> impl Filter<Extract = (Box<dyn Reply>,), Error = Rejection> + Clone {
+    warp::get()
+        .and(warp::path(".well-known"))
+        .and(warp::path("apollo"))
+        .and(warp::path("server-health"))
```

Comment:
Fwiw, this `server` name in particular is a relic of the actual health-check in Apollo Gateway being powered by the underlying Apollo Server.

I low-key wonder if we should use a different name.  (i.e., _someone might just ask, why not `router-health`, because that's what it is, ya?_).  But ultimately doesn't matter all that much right now.

Reply:
Changing the name might be a good idea, but if we did we should keep the name the same in both products. Just because it makes substitution of the router for the gateway easier.

Reply:
(My 2 cents on this: I don't think gateway and router are swappable because the configuration file format is already different)

Reply:
But it still makes substitution easier.

Reply:
The healthcheck is set in the dockerfile anyway

Reply:
Conceivably there could be three `server-health`, `gateway-health`, `router-health`, but I'm unconvinced it adds any value.  We can't change the health check for all the existing Gateways without it being a breaking change, and doesn't seem justifiable.  If it was `gateway-health`, I'd actually think it was more compelling of a reason to change it to stay on brand but the router is still a server, so let it keep the same health check path. 😉 

I'm going to mark this (my) thread as resolved here.